src/ctap/client_pin.rs

```diff
@@ -14,141 +14,90 @@
 
 use super::command::AuthenticatorClientPinParameters;
 use super::data_formats::{ClientPinSubCommand, CoseKey, GetAssertionHmacSecretInput};
+use super::pin_protocol::{verify_pin_uv_auth_token, PinProtocol, SharedSecret};
 use super::response::{AuthenticatorClientPinResponse, ResponseData};
 use super::status_code::Ctap2StatusCode;
 use super::storage::PersistentStore;
 use alloc::str;
 use alloc::string::String;
-use alloc::vec;
 use alloc::vec::Vec;
-use arrayref::array_ref;
-use core::convert::TryInto;
-use crypto::cbc::{cbc_decrypt, cbc_encrypt};
-use crypto::hmac::{hmac_256, verify_hmac_256_first_128bits};
+use crypto::hmac::hmac_256;
 use crypto::rng256::Rng256;
 use crypto::sha256::Sha256;
 use crypto::Hash256;
 #[cfg(test)]
 use enum_iterator::IntoEnumIterator;
 use subtle::ConstantTimeEq;
 
-// Those constants have to be multiples of 16, the AES block size.
+/// The prefix length of the PIN hash that is stored and compared.
+///
+/// The code assumes that this value is a multiple of the AES block length and
+/// fits an u8.
 pub const PIN_AUTH_LENGTH: usize = 16;
-const PIN_PADDED_LENGTH: usize = 64;
-const PIN_TOKEN_LENGTH: usize = 32;
 
-/// Checks the given pin_auth against the truncated output of HMAC-SHA256.
-/// Returns LEFT(HMAC(hmac_key, hmac_contents), 16) == pin_auth).
-fn verify_pin_auth(hmac_key: &[u8], hmac_contents: &[u8], pin_auth: &[u8]) -> bool {
-    if pin_auth.len() != PIN_AUTH_LENGTH {
-        return false;
-    }
-    verify_hmac_256_first_128bits::<Sha256>(
-        hmac_key,
-        hmac_contents,
-        array_ref![pin_auth, 0, PIN_AUTH_LENGTH],
-    )
-}
+/// The length of the pinUvAuthToken used throughout PIN protocols.
+pub const PIN_TOKEN_LENGTH: usize = 32;
```

Comment:
This values are not customizable and tests break if we change them because the crypto does not match anymore. I updated the documentation to be more precise about internal and external constraints.